tests/framework/prototype.py

```diff
@@ -244,6 +245,37 @@ def assert_rule(self, custom_resource=None, cr_namespace='default'):
         pod_exec_command(headpod_name, cr_namespace,
             "python -c \"import ray; ray.init(); print(ray.cluster_resources())\"")
 
+class ShutdownJobRule(Rule):
+    """Check the Ray cluster is shutdown when setting `spec.shutdownAfterJobFinishes` to true."""
+    def assert_rule(self, custom_resource=None, cr_namespace='default'):
+        custom_api = client.CustomObjectsApi()
```

Comment:
Multiple K8s API client instances will introduce unstability. You can reuse `K8S_CLUSTER_MANAGER.k8s_client_dict[CONST.K8S_CR_CLIENT_KEY]` instead.

Reply:
Oh I didn't know this, thanks for the heads up!